src/app/static/src/js/components/runWorkflow/runWorkflowProgress.vue

```diff
@@ -1,19 +1,206 @@
+import ErrorInfo from "../errorInfo.vue";
+import {
+    RunWorkflowMetadata,
+    WorkflowRunSummary,
+    WorkflowRunStatus,
+} from "../../utils/types";
+import { buildFullUrl } from "../../utils/api";
+import { session } from "./../../utils/session.js";
+
+interface Data {
+    workflowRunSummaries: null | WorkflowRunSummary[];
+    selectedWorkflowKey: null | string;
+    workflowRunStatus: null | WorkflowRunStatus;
+    error: string;
+    defaultMessage: string;
+}
+
+interface Methods {
+    getWorkflowRunSummaries: () => void;
+    getWorkflowRunStatus: (key: string) => void;
+    formatDate: (date: string) => string;
+    reportVersionHref: (name: string, version: string) => string;
+    reportLogsHref: (name: string) => string;
+    statusColour: (status: string) => string;
+    interpretStatus: (status: string) => string;
+    setReportLogsTab: () => void;
+}
 
 interface Props {
-    workflowMetadata: RunWorkflowMetadata | null
+    workflowMetadata: RunWorkflowMetadata | null;
 }
 
-export default Vue.extend<unknown, unknown, unknown, Props>({
+export default Vue.extend<Data, Methods, unknown, Props>({
     name: "runWorkflowProgress",
+    components: {
+        ErrorInfo,
+        vSelect,
+    },
     props: {
-        workflowMetadata: null
-    }
-})
+        workflowMetadata: null,
+    },
+    data() {
+        return {
+            workflowRunSummaries: null,
+            selectedWorkflowKey: null,
+            workflowRunStatus: null,
+            error: "",
+            defaultMessage: "",
+        };
+    },
+    methods: {
+        getWorkflowRunSummaries() {
+            api.get("/workflows")
+                .then(({ data }) => {
+                    this.workflowRunSummaries = data;
+                    this.error = "";
+                    this.defaultMessage = "";
+                })
+                .catch((error) => {
+                    this.error = error;
+                    this.defaultMessage =
+                        "An error occurred fetching the workflows";
+                });
+        },
+        getWorkflowRunStatus(key) {
+            api.get(`/workflows/${key}/status`)
+                .then(({ data }) => {
+                    this.workflowRunStatus = data;
+                    this.error = "";
+                    this.defaultMessage = "";
+                })
+                .catch((error) => {
+                    this.error = error;
+                    this.defaultMessage =
+                        "An error occurred fetching the workflow reports";
+                });
+        },
+        formatDate(date) {
+            return longTimestamp(new Date(date));
+        },
+        reportVersionHref(name, version) {
+            const url = `/report/${name}/${version}/`;
+            return buildFullUrl(url);
+        },
+        reportLogsHref(name) {
+            const url = `/run-report?report-name=${name}`; // full implementation into report logs will be covered in a future ticket
```

Comment:
If the name contains spaces, will that be automatically fused when it is made into a url or do I have to do that here?

Reply:
The name [shouldn't include spaces](https://www.vaccineimpact.org/orderly/articles/orderly.html#creating-a-report) and I don't think it will include any characters needing URL-encoding so I think you're fine here

Reply:
My memory is failing me as this was so long ago but I believe I meant that run reports would handle the url params to bring up the appropriate report logs. This is alluded to in the mock ups.